### PR TITLE
Add secrets model with encrypted value storage

### PR DESCRIPTION
**Pull request by suricactus**

### CODE REVIEW

**Comment by duke-nyuki** [comment]
Task linked: [QF-117 Manage remote db passwords on server](https://app.clickup.com/t/2192114/QF-117)

File: docker-app/worker_wrapper/wrapper.py [hunk 1]
```diff
@@ -203,13 +204,26 @@ def _run_docker(
 
         client = docker.from_env()
 
+        extra_envvars = {}
+        pgservice_file_contents = ""
+        for secret in self.job.project.secrets.all():
+            if secret.type == Secret.Type.ENVVAR:
+                extra_envvars[secret.name] = secret.value
+            elif secret.type == Secret.Type.PGSERVICE:
+                pgservice_file_contents += f"\r\n{secret.value}"
```
Comment (olivierdalang) [inline docker-app/worker_wrapper/wrapper.py:213]: Are you sure about the windows line ending ? Does a test case cover multiple secrets ?
```
[conf1]
host=127.0.0.1
...
password=postgres\r
[conf2]
host=127.0.0.1
...
```

**Review by olivierdalang — COMMENTED** [review]
Ok review done

File: docker-app/qfieldcloud/core/models.py [hunk 2]
```diff
@@ -1239,6 +1240,39 @@ def __str__(self):
...
+        ENVVAR = "envvar", _("Environment Variable")
+
+    name = models.TextField(
+        max_length=255,
+        unique=True,
+        validators=[
+            RegexValidator(
+                r"^[A-Z]+[A-Z0-9_]+$",
+                _(
+                    "Must start with a letter and followed by capital letters, numbers or underscores."
```
[the remark this answers is not in the document]
Comment (suricactus) [reply docker-app/qfieldcloud/core/models.py:1255]: done!

File: docker-app/qfieldcloud/core/models.py [hunk 3]
```diff
@@ -1239,6 +1240,39 @@ def __str__(self):
...
+                ),
+            )
+        ],
+        help_text=_(
+            _(
+                "Must start with a letter and followed by capital letters, numbers or underscores."
```
[the remark this answers is not in the document]
Comment (suricactus) [reply docker-app/qfieldcloud/core/models.py:1261]: done!

File: docker-app/qfieldcloud/core/models.py [hunk 4]
```diff
@@ -1239,6 +1240,39 @@ def __str__(self):
...
 
 
+class Secret(models.Model):
+    class Type(models.TextChoices):
+        PGSERVICE = "pgservice", _("pg_service")
+        ENVVAR = "envvar", _("Environment Variable")
+
+    name = models.TextField(
+        max_length=255,
+        unique=True,
```
[the remark this answers is not in the document]
Comment (suricactus) [reply docker-app/qfieldcloud/core/models.py:1250]: Good catch.

File: docker-app/qfieldcloud/core/models.py [hunk 5]
```diff
@@ -1239,6 +1240,39 @@ def __str__(self):
...
+        ),
+    )
+    type = models.CharField(max_length=32, choices=Type.choices)
+    project = models.ForeignKey(
+        Project, on_delete=models.CASCADE, related_name="secrets"
+    )
+    created_by = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="project_secrets"
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
```
[the remark this answers is not in the document]
Comment (suricactus) [reply docker-app/qfieldcloud/core/models.py:1272]: They are not updatable, that's why I didn't add them.

File: docker-app/worker_wrapper/wrapper.py (hunk 1 above)
Comment (suricactus) [reply docker-app/worker_wrapper/wrapper.py:213]: It doesn't. Good idea.

**Pull request merged**